### ISSUE TITLE
Cannot install cLoops2 as instructed

### ISSUE THREAD

**Issue by CharlsW2023** [YaqiangCao/cLoops2#8]
Hi,

I tried to install cLoops2 from source for latest version:

$ conda create -n cLoops2-2 python=3.7
$ conda activate cLoops2-2
$ conda env update --file cLoops2_env.yaml

It shows errors:

Collecting package metadata (repodata.json): done
Solving environment: done

Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Installing pip dependencies: / Ran pip subprocess with arguments:
['/home/mypath/anaconda3/envs/cLoops2-2/bin/python', '-m', 'pip', 'install', '-U', '-r', '/scr1/users/mypath/Software2/cLoops2/condaenv.emnigv1o.requirements.txt', '--exists-action=b']
Pip subprocess output:

Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement cloops2==1.0.0 (from -r /scr1/users/mypath/Software2/cLoops2/condaenv.emnigv1o.requirements.txt (line 1)) (from versions: 0.0.2, 0.0.3)
ERROR: No matching distribution found for cloops2==1.0.0 (from -r /scr1/users/mypath/Software2/cLoops2/condaenv.emnigv1o.requirements.txt (line 1))

failed

CondaEnvException: Pip failed

----------------------------------------------------
I then tried to install cLoops2 directly from pip:
$ conda deactivate
$ pip install cLoops2
It says:
Successfully installed cLoops2-0.0.3 pyBigWig-0.3.22

But when I run $ cLoops2
It shows error:

Traceback (most recent call last):
  File "/cm/shared/easybuild/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages/numpy/core/__init__.py", line 23, in <module>
    from . import multiarray
  File "/cm/shared/easybuild/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages/numpy/core/multiarray.py", line 10, in <module>
    from . import overrides
  File "/cm/shared/easybuild/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages/numpy/core/overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mypath/anaconda3/bin/cLoops2", line 5, in <module>
    from cLoops2.cLoops2 import main
  File "/home/mypath/anaconda3/lib/python3.9/site-packages/cLoops2/cLoops2.py", line 25, in <module>
    import numpy as np
  File "/cm/shared/easybuild/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages/numpy/__init__.py", line 144, in <module>
    from . import core
  File "/cm/shared/easybuild/software/SciPy-bundle/2022.05-foss-2022a/lib/python3.10/site-packages/numpy/core/__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "/home/mypath/anaconda3/bin/python"
  * The NumPy version is: "1.22.3"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

 
Looking forward to your reply!

Thanks!

Best,
Charles

**Comment by CharlsW2023** [comment]
I then modify the  cLoops2_env.yaml to not request pip install cLoops2==1.0.0 version (remove ==1.0.0) and run
$ conda create -n cLoops2-2 python=3.7
$ conda activate cLoops2-2
$ conda env update --file cLoops2_env.yaml

The error shows:

Pip subprocess error:
ERROR: Package 'SciPy' requires a different Python: 3.7.0 not in '>=3.8,<3.11'

failed

**Comment by YaqiangCao** [comment]
> I then modify the cLoops2_env.yaml to not request pip install cLoops2==1.0.0 version (remove ==1.0.0) and run $ conda create -n cLoops2-2 python=3.7 $ conda activate cLoops2-2 $ conda env update --file cLoops2_env.yaml
> 
> The error shows:
> 
> Pip subprocess error: ERROR: Package 'SciPy' requires a different Python: 3.7.0 not in '>=3.8,<3.11'
> 
> failed

Hi Charles, 
    Sorry for the late reply. Have you solved the problem?  
    I could recommend you to install the following 3rd dependencies with conda or pip, then in the cLoops2 directory run python setup.py install to install it. 
```
tqdm
numpy 
scipy 
pandas
sklearn
seaborn
pyBigWig
matplotlib
joblib
networkx
```

Best,
Yaqiang

**Comment by CW20221031** [comment]
Hi Yaqiang,

Thank you for your reply!

I found the issue and solved it now. The issue is that I added "module load Bowtie2" in my .bashrc to load the pre-installed Bowtie2 module in our cluster, which would auto-load Python3.10 in our cluster. And this is conflicted with my conda environment. Especially if the packages I want to install in my conda environment are already installed in our cluster's Python3.10 directory, it would display Requirement already satisfied: xxx package in our cluster's Python3.10 directory. But it cannot actually use these packages (I don't know why) when running cLoops2.

To solve this, I need to run "module purge" first to remove all modules I loaded. And then run pip install cLoops2 in my conda environment. I have done debugging for several days and found the issue. Glad it works!

Best,
Charles
